include/zenoh-pico/transport/transport.h

```diff
@@ -60,17 +52,33 @@ typedef struct {
     // Patch
     uint8_t _patch;
 #endif
-} _z_transport_peer_entry_t;
-
-size_t _z_transport_peer_entry_size(const _z_transport_peer_entry_t *src);
-void _z_transport_peer_entry_clear(_z_transport_peer_entry_t *src);
-void _z_transport_peer_entry_copy(_z_transport_peer_entry_t *dst, const _z_transport_peer_entry_t *src);
-bool _z_transport_peer_entry_eq(const _z_transport_peer_entry_t *left, const _z_transport_peer_entry_t *right);
-_Z_ELEM_DEFINE(_z_transport_peer_entry, _z_transport_peer_entry_t, _z_transport_peer_entry_size,
-               _z_transport_peer_entry_clear, _z_transport_peer_entry_copy, _z_noop_move)
-_Z_LIST_DEFINE(_z_transport_peer_entry, _z_transport_peer_entry_t)
-_z_transport_peer_entry_list_t *_z_transport_peer_entry_list_insert(_z_transport_peer_entry_list_t *root,
-                                                                    _z_transport_peer_entry_t *entry);
+} _z_transport_peer_common_t;
+
+void _z_transport_peer_common_clear(_z_transport_peer_common_t *src);
+void _z_transport_peer_common_copy(_z_transport_peer_common_t *dst, const _z_transport_peer_common_t *src);
+bool _z_transport_peer_common_eq(const _z_transport_peer_common_t *left, const _z_transport_peer_common_t *right);
+
+typedef struct {
+    _z_slice_t _remote_addr;
+    _z_conduit_sn_list_t _sn_rx_sns;
+    // SN numbers
+    _z_zint_t _sn_res;
+    volatile _z_zint_t _lease;
+    volatile _z_zint_t _next_lease;
+    uint16_t _peer_id;
+    _z_transport_peer_common_t common;
```

Comment:
Is there any value in having the common member first to ensure a similar memory layout between structs containing the same information?